src/Entity.js

```diff
@@ -0,0 +1,359 @@
+import Action from './Action.js';
+import assert from './assert.js';
+import Link from './Link.js';
+import { contains, getMatchingValue, getMatchingValuesByAll, hasProperty } from './util.js';
+
+export default class Entity {
```

Comment:
Not actually new, I just moved this from `index.js` into a proper `Entity.js`. Ultimately this is generally the entrypoint here, but not always.